### PR TITLE
Properly check for OS X sysroot

### PR DESCRIPTION
**Pull request by mkuron**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -171,6 +171,10 @@ if (WITH_CUDA)
         list(APPEND CUDA_NVCC_FLAGS "-D_MWAITXINTRIN_H_INCLUDED -D_FORCE_INLINES")
       endif()
       SET(CUDA_PROPAGATE_HOST_FLAGS OFF)
+
+      if (CMAKE_OSX_SYSROOT)
+        set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler -isysroot -Xcompiler ${CMAKE_OSX_SYSROOT}")
```
Comment (KaiSzuttor) [inline CMakeLists.txt:176]: why is there the second Xcompiler argument: shouldn't it be:
``` cmake
-Xcompiler \"-isysroot ${CMAKE_OSX_SYSROOT}\"
```

Comment (mkuron) [reply CMakeLists.txt:176]: The way I have it is correct. If you put it in quotes, the space actually gets passed to the compiler.

Comment (KaiSzuttor) [reply CMakeLists.txt:176]: this is what you want, right? the argument to clang should be ``-isysroot /path/to/root``

Comment (mkuron) [reply CMakeLists.txt:176]: No. Clang needs two separate arguments, one `-isysroot`, one `/path/to/root`. Trust me, this is correct and has been working for years.

**Pull request merged**